File: OpenCV/Face_recognition_system/FRS.py
import numpy as np
import cv2
import time

cam = cv2.VideoCapture(0)

if not cam.isOpened():
    print("Camera is not available")
    exit()

try:
    cam.set(cv2.CAP_PROP_FRAME_WIDTH, 1280)
    cam.set(cv2.CAP_PROP_FRAME_HEIGHT, 720)

except:
    print("Camera does not support 720p")

width = int(cam.get(cv2.CAP_PROP_FRAME_WIDTH))  
height = int(cam.get(cv2.CAP_PROP_FRAME_HEIGHT))
fps = int(cam.get(cv2.CAP_PROP_FPS))
print("{}x{} {}fps".format(width, height, fps))

upscale = 4.0
downscale = 1/4

while True:
    ret, img = cam.read()
    if not ret:
        print("Camera is not available")
        break

    cv2.imshow('FRS', img)
    
    img = cv2.resize(img, (0, 0), fx=upscale, fy=upscale)
    kernel = np.array([[-1,-1,-1], [-1,9,-1], [-1,-1,-1]])  
    img = cv2.filter2D(img, -1, kernel)
    img = cv2.resize(img, (0, 0), fx=downscale, fy=downscale)
    cv2.imshow('FRS enhanced', img) 

    key = cv2.waitKey(1)    
    if key == 27:    #ESC
        break   

    # elif key == 13:    #Enter
    #     file_name = ("ScreenShot "+time.ctime(time.time())+ ".jpeg")
    #     image = cv2.imwrite(file_name, img)  

cv2.destroyAllWindows()

File: OpenCV/Face_recognition_system/FRS.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
#include <ctime>

int main(int argc, char** argv)
{
    cv::VideoCapture cap(0); // open the default camera
    if(!cap.isOpened())  // check if we succeeded
    {
        std::cout << "Camera is not available" << '\n';
        return -1;
    }

    int width = cap.get(cv::CAP_PROP_FRAME_WIDTH);
    int height = cap.get(cv::CAP_PROP_FRAME_HEIGHT);
    int fps = cap.get(cv::CAP_PROP_FPS);
    std::cout << width << "x" << height << " " << fps << "fps" << '\n';

    cv::namedWindow("FRS",1);
    cv::namedWindow("FRS enhanced",1);
    cv::Mat img, img_enhanced;

    for(;;)
    {
        cap >> img; // get a new frame from camera
        if(img.empty())
        {
            std::cout << "Camera is not available" << '\n';
            break;
        }

        cv::imshow("FRS", img);

        cv::Mat kernel = (cv::Mat_<float>(3,3) << -1,-1,-1, -1,9,-1, -1,-1,-1);
        cv::filter2D(img, img_enhanced, -1 , kernel);
        cv::imshow("FRS enhanced", img_enhanced);

        char key = (char)cv::waitKey(1);
        if(key == 27) // ESC
            break;
        else if(key == 13) // Enter
        {
            time_t now = time(0);
            tm *ltm = localtime(&now);
            std::string file_name = "ScreenShot " + std::to_string(ltm->tm_hour) + ":" + std::to_string(ltm->tm_min) + ":" + std::to_string(ltm->tm_sec) + ".jpeg";
            cv::imwrite(file_name, img_enhanced);
        }
    }

    // the camera will be deinitialized automatically in VideoCapture destructor
    return 0;
}
